sphinx_source/Ucuncu_taraf_moduller/django_modülü/django_2.rst

```diff
@@ -0,0 +1,487 @@
+    ]
+
+Artık Django, polls isimli uygulamamızı projeye eklediğimizi biliyor. 
+Şimdi farklı bir komutu çalıştıralım::
+
+    python manage.py makemigrations polls
+
+Talimatlara benzer şeyler görmelisiniz::
+
+    Migrations for 'polls':
+      polls/migrations/0001_initial.py:
+        - Create model Choice
+        - Create model Question
+        - Add field question to choice
+
+`makemigrations` komutu ile ürettiğimiz modellerde değişiklik 
+yaptığımızı (Aynı zamanda yeni modeller ürettiğimizi de) ve 
+değişikliklerin migration olarak depolanmasını istediğimizi 
+söyledik. Migrationlar senin modellerindeki değişiklerin 
+depolanma şeklidir(ve bu yüzden veritabanı şeması). 
+İstersen yeni modelinin migrationunu okuyabilirsin. 
+Migration **polls/migrations/0001_inital.py** isimli dosyada. 
+Endişelenmeyin, bu dosyayı her zaman okumak zorunda değilsiniz.  
+"`migrate`" komutu, senin veritabanı 
+şemanı otomatik olarak yönetmek için  migrationları çalıştırır.
+Birazdan bu konuya tekrar geleceğiz. Ama önce migrationın çalıştırdığı 
+SQL dosyasına bakalım. "`sqlmigrate`" komutuna migration adını 
+parametre olarak verin ve SQL çıktısını alın::
+
+    python manage.py sqlmigrate polls 0001
+
+İngilizce yönerge tarzında bir şeyler görüyor olmalısın::
+
+    BEGIN;
+    --
+    -- Create model Choice
+    --
+    CREATE TABLE "polls_choice" (
+        "id" serial NOT NULL PRIMARY KEY,
+        "choice_text" varchar(200) NOT NULL,
+        "votes" integer NOT NULL
+    );
+    --
+    -- Create model Question
+    --
+    CREATE TABLE "polls_question" (
+        "id" serial NOT NULL PRIMARY KEY,
+        "question_text" varchar(200) NOT NULL,
+        "pub_date" timestamp with time zone NOT NULL
+    );
+    --
+    -- Add field question to choice
+    --
+    ALTER TABLE "polls_choice" ADD COLUMN "question_id" integer NOT NULL;
+    ALTER TABLE "polls_choice" ALTER COLUMN "question_id" DROP DEFAULT;
+    CREATE INDEX "polls_choice_7aa0f6ee" ON "polls_choice" ("question_id");
+    ALTER TABLE "polls_choice"
+    ADD CONSTRAINT "polls_choice_question_id_246c99a640fbbd72_fk_polls_question_id"
+        FOREIGN KEY ("question_id")
+        REFERENCES "polls_question" ("id")
+        DEFERRABLE INITIALLY DEFERRED;
+
+    COMMIT;
+
+Aşağıdakilere dikkat et:
+    - Çıktı kullandığınız veritabanına göre değişir. 
+      Yukarıdaki örnek PostgreSQL için yazılmıştır.
+    - Tablo adı, uygulamanın adını ve modelin adını
+      (küçük harflerle) kullanarak otomatik olarak oluşturulur .
+      (Bu davranış geçersiz kılınabilir.)
+    - Birincil anahtarlar(ID) otomatik olarak eklenir.
+      (Bu da geçersiz kılınabilir.)
+    - Düzenlenirken  Django foreign key alanının adına "_id" 
+      ekler.(Evet, bunu da geçersiz kılabilirsiniz.)
+    - Yabancı anahtar ilişkisi bir FOREIGN KEY kısıtlaması 
+      ile açık hale getirilmiştir. DEFERRABLE parçaları hakkında 
+      endişelenmeyin; Bu sadece PostgreSQL'in yabancı anahtarını 
```

Comment:
"yabancı anahtarını" -> "yabancı anahtarı"
Bu şekilde mi olmalı acaba?

Reply:
İkisinin benim için bir farkı yok. Ancak madem takıldınız değiştirelim.